### DISCUSSION TITLE
The problem of repeated re-rendering

### DISCUSSION THREAD

**Discussion by Keberson** [annotorious/annotorious#453] [General]
My code is re-rendered 4 times to display a single image, hence the problem that popup throws an error in an attempt to render. If I try to open the annotation, I run into a problem - Rendered more hooks than during the previous render.
Maybe I'm doing something wrong?

https://codesandbox.io/p/github/Keberson/annotoriuos/master?import=true

**Comment by rsimon** [comment]
Please provide a reproducible example via CodeSandbox, StackBlitz etc.

**Reply by Keberson** [reply]
https://codesandbox.io/p/github/Keberson/annotoriuos/master?import=true

If you will try to click on annotation, you will get an error

**Comment by GiuseppeSpathis** [comment]
Try to remove react strict mode
import React from 'react';
import ReactDOM from 'react-dom';

ReactDOM.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>,
  document.getElementById('root')
);

Remove <React.StrictMode>

**Reply by Keberson** [reply]
Already deleted ((
You can check my code

https://codesandbox.io/p/github/Keberson/annotoriuos/master?import=true

**Comment by rsimon** [comment]
Did you read the example in the docs?

https://annotorious.dev/react/component-reference/#openseadragonpopup

If I make this change, things seem to work fine.

```tsx
<OpenSeadragonAnnotationPopup
    popup={props => (
        <CommentPopup {...props} />
    )} />
```

**Reply by Keberson** [reply]
Yes, you are right/. Thx a lot!

I read all examples, but I relied on this example: https://annotorious.dev/react/custom-popups/#example-editable-comments
This was my big error

**Reply by rsimon** [reply]
Ah - thanks for pointing out. Yes, I believe this example is actually wrong - will fix!
